scripts/js/lib/api/objectsInv.ts

```diff
@@ -110,43 +110,61 @@ export class ObjectsInv {
     for (const line of lines) {
-      // Regex from sphinx source
-      // https://github.com/sphinx-doc/sphinx/blob/2f60b44999d7e610d932529784f082fc1c6af989/sphinx/util/inventory.py#L115-L116
-      const parts = line.match(/(.+?)\s+(\S+)\s+(-?\d+)\s+?(\S*)\s+(.*)/);
-      if (parts == null || parts.length != 6) {
-        console.warn(`Error parsing line of objects.inv: ${line}`);
-        continue;
-      }
-      const entry = {
-        name: parts[1],
-        domainAndRole: parts[2],
-        priority: parts[3],
-        uri: parts[4],
-        dispname: parts[5],
-      };
-      entry.uri = ObjectsInv.#expandUri(entry.uri, entry.name);
-      if (!shouldIncludeEntry(entry, packageLanguage)) {
-        continue;
-      }
-      if (packageLanguage === "C") {
-        entry.uri = ObjectsInv.#transformCApiUri(entry.uri, entry.name);
-      }
-
-      entries.push(entry);
+      const entry = ObjectsInv.lineToEntry(line, packageLanguage);
+      if (entry) entries.push(entry);
     }
 
     return new ObjectsInv(preamble, entries);
   }
 
+  public static lineToEntry(
+    line: string,
+    packageLanguage: PackageLanguage | "any",
+  ): ObjectsInvEntry | null {
+    // Regex from sphinx source
+    // https://github.com/sphinx-doc/sphinx/blob/2f60b44999d7e610d932529784f082fc1c6af989/sphinx/util/inventory.py#L115-L116
+    const parts = line.match(/(.+?)\s+(\S+)\s+(-?\d+)\s+?(\S*)\s+(.*)/);
+    if (parts == null || parts.length != 6) {
+      console.warn(`Error parsing line of objects.inv: ${line}`);
+      return null;
+    }
+    const entry = {
+      name: parts[1],
+      domainAndRole: parts[2],
+      priority: parts[3],
+      uri: parts[4],
+      dispname: parts[5],
+    };
+    entry.uri = ObjectsInv.#expandUri(entry.uri, entry.name);
+    if (!shouldIncludeEntry(entry, packageLanguage)) {
+      return null;
+    }
+    if (packageLanguage === "C") {
+      entry.uri = ObjectsInv.transformCApiUri(entry.uri, entry.name);
+    }
+    return entry;
+  }
+
   /**
    * The anchors from the C API artifact do not make it into the final page, so
    * we must manipulate the URI based on the object name.
    */
-  static #transformCApiUri(uri: string, name: string): string {
+  public static transformCApiUri(uri: string, name: string): string {
     const uriWithoutPrefix = removePrefix(uri, C_API_BASE_PATH + "/");
     if (!uriWithoutPrefix.includes("#")) return uriWithoutPrefix;
     const [path, _anchor] = uriWithoutPrefix.split("#");
 
+    // We sometimes get anchors of the form
+    // 'structQkObsTerm_1a14ff1665641903565439ad9877fd2c8e' and
+    // 'structQkObsTerm_1autotoc_md2', we just map these to the object.
+    const maybeStructAutoTocMatch =
+      uri.match(/struct([A-z]+)_(\dautotoc_|[\dabcdef]{34})/) ||
+      uri.match(/struct([A-Z][A-z]+)$/);
+    if (maybeStructAutoTocMatch) {
+      const objectName = maybeStructAutoTocMatch[1].toLowerCase();
+      return `${path}#${objectName}`;
+    }
```

Comment:
This is best-effort, it seems to work with the current C API, but we can adjust if we hit more edge cases. This should not affect the Python API.